core/src/main/scala/org/apache/spark/ui/exec/ExecutorThreadDumpPage.scala

```diff
@@ -48,7 +49,13 @@ private[ui] class ExecutorThreadDumpPage(parent: ExecutorsTab) extends WebUIPage
     val time = System.currentTimeMillis()
     val maybeThreadDump = sc.get.getExecutorThreadDump(executorId)
 
-    val content = maybeThreadDump.map { threadDump =>
+    val grepExp = Option(request.getParameter("grepexp")).map(Pattern.compile(_, Pattern.MULTILINE))
+
+    val filteredContent = maybeThreadDump.map { threadDump =>
+      threadDump.filter(thread =>
+        grepExp.map { grep => grep.matcher(thread.stackTrace).find()}.getOrElse(true)
+      )
```

Comment:
so for these, I think we are going to port the grep implementation to javascript, right?